src/Network/Network/AzureFirewall/NewAzureFirewallCommand.cs

```diff
@@ -154,6 +154,26 @@ public class NewAzureFirewallCommand : AzureFirewallBaseCmdlet
         )]
         public string[] PrivateRange { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Flag to indicate DNS Proxy is enabled or disabled"
+        )]
+        [ValidateSet("true", "false", IgnoreCase = false)]
+        public string DNSEnableProxy { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Flag to indicate if DNS Proxy is required for Network Rules"
+        )]
+        [ValidateSet("true", "false", IgnoreCase = true)]
+        public string DNSRequireProxyForNetworkRules { get; set; }
```

Comment:
Smae thing about switch parameter.  Also, please do not use plural name as parameter name in powershell. It's discouraged. See https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/parameter-best-practices.md#singularity

Reply:
Done